pkg/backend/display/rows.go

```diff
@@ -457,45 +470,49 @@ func getDiffInfo(step engine.StepEventMetadata, action apitype.UpdateKind) strin
 		recordMetadataDiff("protect",
 			resource.NewBoolProperty(step.Old.Protect), resource.NewBoolProperty(step.New.Protect))
 
-		if diff != nil {
-			writeString(changesBuf, "diff: ")
+		writeShortDiff(changesBuf, diff, step.Diffs)
+	}
 
-			updates := make(resource.PropertyMap)
-			for k := range diff.Updates {
-				updates[k] = resource.PropertyValue{}
-			}
+	fprintIgnoreError(changesBuf, colors.Reset)
+	return changesBuf.String()
+}
 
-			filteredKeys := func(m resource.PropertyMap) []string {
-				keys := slice.Prealloc[string](len(m))
-				for k := range m {
-					keys = append(keys, string(k))
-				}
-				return keys
+func writeShortDiff(changesBuf io.StringWriter, diff *resource.ObjectDiff, include []resource.PropertyKey) {
+	if diff != nil {
+		writeString(changesBuf, "diff: ")
+
+		updates := make(resource.PropertyMap)
+		for k := range diff.Updates {
+			updates[k] = resource.PropertyValue{}
+		}
+
+		filteredKeys := func(m resource.PropertyMap) []string {
+			keys := make([]string, 0, len(m))
```

Comment:
Any reason to move away from `keys := slice.Prealloc[string](len(m))`? (Or was it just a bad merge?)

Reply:
Bad rebase. Fixed.